server/manager/src/main/java/org/apache/accumulo/manager/Manager.java

```diff
@@ -425,6 +428,7 @@ protected Manager(ServerOpts opts, String[] args) throws IOException {
     log.info("Instance {}", getInstanceID());
     timeKeeper = new ManagerTime(this, aconf);
     tserverSet = new LiveTServerSet(context, this);
+    context.getPropStore().registerAsListener(SystemPropKey.of(context), this);
```

Comment:
The impl of `registerAsListener` will add the listener to a set, so the order in which listeners are called will vary at runtime.  I am not sure, but it seems like maybe there is another listener that invalidates the prop store cache. If this is the case this the listener registered here could execute before or after the listener that clears the cache depending on their order in the hash set.   That could cause this listener to sometime see config change and sometime now.  However this may not be correct, drawing this conclusion base on looking at some of the code.  If it is the case maybe a LinkedHashSet could be used to make the order in which listeners are called correspond with the order they were added.